wallet-sdk/src/main/kotlin/org/stellar/walletsdk/auth/AuthHeaderSigner.kt

```diff
@@ -0,0 +1,106 @@
+package org.stellar.walletsdk.auth
+
+import io.jsonwebtoken.JwtBuilder
+import io.jsonwebtoken.Jwts
+import io.ktor.client.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.serialization.kotlinx.json.*
+import java.time.Instant
+import java.util.*
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.minutes
+import kotlinx.datetime.Clock
+import kotlinx.serialization.Serializable
+import org.stellar.walletsdk.horizon.AccountKeyPair
+import org.stellar.walletsdk.horizon.SigningKeyPair
+import org.stellar.walletsdk.util.Util.postJson
+import org.stellar.walletsdk.util.toJava
+
+/** Header signer to sign JWT for GET /Auth request. */
+interface AuthHeaderSigner {
+  suspend fun createToken(
+    claims: Map<String, String>,
+    clientDomain: String?,
+    issuer: AccountKeyPair?
+  ): String
+}
+
+/** Header signer signing JWT for GET /Auth with a main custodial key */
+open class DefaultAuthHeaderSigner(open val expiration: Duration = 15.minutes) : AuthHeaderSigner {
+  override suspend fun createToken(
+    claims: Map<String, String>,
+    clientDomain: String?,
+    issuer: AccountKeyPair?
+  ): String {
+    require(issuer != null) { "Default signer can't sign headers for client domain." }
+    require(issuer is SigningKeyPair) {
+      "SigningKeyPair must be provided to .auth() method in order to sign headers"
+    }
+
+    val timeExp = Instant.ofEpochSecond(Clock.System.now().plus(expiration).epochSeconds)
+    val builder = createBuilder(timeExp, claims)
+
+    builder.signWith(issuer.toJava().private, Jwts.SIG.EdDSA)
+
+    return builder.compact()
+  }
+
+  fun createBuilder(timeExp: Instant, claims: Map<String, String>): JwtBuilder {
+    return Jwts.builder().issuedAt(Date.from(Instant.now())).expiration(Date.from(timeExp)).also {
+      builder ->
+      claims.forEach { builder.claim(it.key, it.value) }
+    }
+  }
+}
+
+open class DomainAuthHeaderSigner(
```

Comment:
Could we add a comment here like we have for `DefaultAuthHeaderSigner` ?

Reply:
Done

Reply:
Thanks! Have you pushed it? I don't see the change just yet

Reply:
Sorry didn't push 😅